projects/kit/components/input-month/input-month.directive.ts

```diff
@@ -0,0 +1,136 @@
+        '(blur)': 'onTouched()',
+        '[disabled]': 'disabled()',
+        // Make <input /> readonly (without readonly appearance)
+        inputmode: 'none',
+        '[style.caretColor]': '"transparent"',
+        '(beforeinput)': '$event.inputType.includes("delete") || $event.preventDefault()',
+        '(input)': '$event.inputType?.includes("delete") && clear()',
+    },
+})
+export class TuiInputMonthDirective extends TuiControl<TuiMonth | null> {
+    private readonly options = inject(TUI_INPUT_MONTH_OPTIONS);
+    private readonly textfieldOptions = inject(TUI_TEXTFIELD_OPTIONS);
+    private readonly injector = inject(INJECTOR);
+
+    private readonly open = tuiDropdownOpen();
+
+    private readonly formatter = toSignal(inject(TUI_MONTH_FORMATTER), {
+        initialValue: () => '',
+    });
+
+    protected readonly textfieldValue = tuiValueBinding(
+        computed(() => this.formatter()(this.value())),
+    );
+
+    protected readonly icon = tuiDirectiveBinding(
+        TuiIcons,
+        'iconEnd',
+        computed(() => this.options.icon(this.textfieldOptions.size())),
+        {},
+    );
+
+    protected readonly calendarSync = effect(() => {
+        const calendar = this.calendar();
+
+        if (calendar) {
+            calendar.value.set(this.value());
+            calendar.min.set(this.min() ?? TUI_FIRST_DAY); // TODO(v5): remove TUI_FIRST_DAY fallback
+            calendar.max.set(this.max() ?? TUI_LAST_DAY); // TODO(v5): remove TUI_LAST_DAY fallback
+        }
+    }, TUI_ALLOW_SIGNAL_WRITES);
+
+    protected onMonthClickEffect = effect(() => {
+        const subscription = this.calendar()?.monthClick.subscribe((month) => {
+            this.onChange(month);
+            this.open.set(false);
+        });
+
+        return () => subscription?.unsubscribe();
+    });
+
+    public readonly min = signal<TuiMonth | null>(null);
+    public readonly max = signal<TuiMonth | null>(null);
+    public readonly calendar = tuiInjectAuxiliary<TuiCalendarMonth>();
+
+    constructor() {
+        super();
+
+        /**
+         * Update directive props with new defaults before inputs are processed
+         * TODO: find better way to override TuiDropdownFixed host directive from TuiTextfieldComponent
+         */
+        (inject(TUI_DROPDOWN_OPTIONS) as any).limitWidth = 'auto';
+    }
+
+    @Input('min')
+    public set minSetter(x: TuiMonth | null) {
+        this.min.set(x);
+    }
+
+    @Input('max')
+    public set maxSetter(x: TuiMonth | null) {
+        this.max.set(x);
+    }
+
+    public override setDisabledState(): void {
+        super.setDisabledState();
+        this.open.set(false);
+    }
+
+    protected toggleDropdown(): void {
+        if (this.interactive() && !this.nativePicker?.enabled) {
+            this.open.update((x) => !x);
+        }
+    }
+
+    protected clear(): void {
+        this.onChange(null);
+
+        if (this.nativePicker?.enabled) {
+            this.nativePicker.showPicker();
```

Comment:
What if I just want to remove the value – on android it would be a full on modal window. I think it's ok not to toggle it automatically on cleaner tap

Reply:
fixed in https://github.com/taiga-family/taiga-ui/pull/10345/commits/3e93bc54d56b6d8e5a72124aabd2f15c9e111133